analyzers/tests/SonarAnalyzer.UnitTest/Rules/Utilities/TokenTypeAnalyzerTest.ClassifierTestHarness.cs

```diff
@@ -0,0 +1,111 @@
+﻿using System.Text;
+using System.Text.RegularExpressions;
+using Microsoft.CodeAnalysis.Text;
+using Moq;
+using SonarAnalyzer.Protobuf;
+using Match = System.Text.RegularExpressions.Match;
+
+namespace SonarAnalyzer.UnitTest.Rules;
+
+public partial class TokenTypeAnalyzerTest
+{
+    private static class ClassifierTestHarness
+    {
+        private const int TokenAnnotationChars = 4; // [u:]
+        private const int PrefixTokenAnnotationChars = 3; // [u:
+
+        public static void AssertTokenTypes(string code, bool allowSemanticModel = true, bool ignoreCompilationErrors = false)
+        {
+            var (tree, model, expectedTokens) = ParseTokens(code, ignoreCompilationErrors);
+            var root = tree.GetRoot();
+            model = allowSemanticModel ? model : new Mock<SemanticModel>(MockBehavior.Strict).Object; // The Mock will throw if the semantic model was used.
+            var tokenClassifier = new SonarAnalyzer.Rules.CSharp.TokenTypeAnalyzer.TokenClassifier(model, false);
+            var triviaClassifier = new SonarAnalyzer.Rules.CSharp.TokenTypeAnalyzer.TriviaClassifier();
+            expectedTokens.Should().SatisfyRespectively(expectedTokens.Select((Func<ExpectedToken, Action<ExpectedToken>>)(e => CheckClassifiedToken)));
+
+            void CheckClassifiedToken(ExpectedToken expected)
+            {
+                var expectedLineSpan = tree.GetLocation(expected.Position).GetLineSpan();
+                var because = $$"""token with text "{{expected.TokenText}}" at position {{expectedLineSpan}} was marked as {{expected.TokenType}}""";
```

Comment:
```suggestion
                var because = $"""token with text "{expected.TokenText}" at position {expectedLineSpan} was marked as {expected.TokenType}""";
```

Reply:
I think this is a top for how we work as it came up twice now (I think the other time it was Zsolt if I remember correctly). I think we should be consistent and always use double $$ because most of the time we use it in C# code. I prefer consistency over brevity in this case.